aws/credential_cache.go

```diff
@@ -101,6 +101,11 @@ func (p *CredentialsCache) singleRetrieve(ctx context.Context) (interface{}, err
 		return *creds, nil
 	}
 
+	return p.Renew(ctx)
+}
+
+// Renew will fetch a new credential from provider and cache it
+func (p *CredentialsCache) Renew(ctx context.Context) (interface{}, error) {
```

Comment:
I think this change would introduce race conditions with the underlying provider's Retrieve method. The CredentailsCache plays two roles.

1.) caching of credentials so they are not retrieved as often
2.) concurrency safe wrapper around a unsafe provider.

a `Renew` method like this would expose the potentially unsafe provider to race conditions

Reply:
Fair point, do I understand correctly that "p.sf.DoChan" is protecting from race conditions by deduplicating?
what if we add a simple mutex here, would it be acceptable?